deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -189,6 +190,9 @@ public Response create(
         EnvironBean envBean = Utils.getEnvStage(environDAO, envName, stageName);
         authorizer.authorize(sc, new Resource(envBean.getEnv_name(), Resource.Type.ENV), Role.OPERATOR);
         String operator = sc.getUserPrincipal().getName();
+        if(StringUtils.isEmpty(buildId)) {
+            throw new TeletaanInternalException(Response.Status.BAD_REQUEST, "Build id can not be empty.");
```

Comment:
Will deploy-board catch this exception?

Reply:
i removed the `submit` button from the UI if the build id is empty, so at least from deploy-board, this API wont be triggered. adding this exception is to make sure when clients call `create deploy` api, the buildId is set appropriately.